test/unit/ngsi20/general/startup-test.js

```diff
@@ -0,0 +1,147 @@
+
+    describe('When the IoT Agent is started with environment variables', function() {
+        beforeEach(function() {
+            process.env.IOTA_CB_HOST = 'cbhost';
+            process.env.IOTA_CB_PORT = '1111';
+            process.env.IOTA_CB_NGSI_VERSION = 'v2';
+            process.env.IOTA_NORTH_HOST = 'localhost';
+            process.env.IOTA_NORTH_PORT = '2222';
+            process.env.IOTA_PROVIDER_URL = 'prvider:3333';
+            process.env.IOTA_REGISTRY_TYPE = 'mongo';
+            process.env.IOTA_LOG_LEVEL = 'FATAL';
+            process.env.IOTA_TIMESTAMP = true;
+            process.env.IOTA_IOTAM_HOST = 'iotamhost';
+            process.env.IOTA_IOTAM_PORT = '4444';
+            process.env.IOTA_IOTAM_PATH = '/iotampath';
+            process.env.IOTA_IOTAM_PROTOCOL = 'PDI_PROTOCOL';
+            process.env.IOTA_IOTAM_DESCRIPTION = 'The IoTAM Protocol';
+            process.env.IOTA_MONGO_HOST = 'mongohost';
+            process.env.IOTA_MONGO_PORT = '5555';
+            process.env.IOTA_MONGO_DB = 'themongodb';
+            process.env.IOTA_MONGO_REPLICASET = 'customReplica';
+            process.env.IOTA_DEFAULT_RESOURCE = '/iot/custom';
+
+            nock.cleanAll();
+
+            iotamMock = nock('http://iotamhost:4444')
+                .post('/iotampath')
+                .reply(200,
+                    utils.readExampleFile('./test/unit/ngsi20/examples/iotamResponses/registrationSuccess.json'));
+        });
+
+        afterEach(function() {
+            delete process.env.IOTA_CB_HOST;
+            delete process.env.IOTA_CB_PORT;
+            delete process.env.IOTA_CB_NGSI_VERSION;
+            delete process.env.IOTA_NORTH_HOST;
+            delete process.env.IOTA_NORTH_PORT;
+            delete process.env.IOTA_PROVIDER_URL;
+            delete process.env.IOTA_REGISTRY_TYPE;
+            delete process.env.IOTA_LOG_LEVEL;
+            delete process.env.IOTA_TIMESTAMP;
+            delete process.env.IOTA_IOTAM_HOST;
+            delete process.env.IOTA_IOTAM_PORT;
+            delete process.env.IOTA_IOTAM_PATH;
+            delete process.env.IOTA_IOTAM_PROTOCOL;
+            delete process.env.IOTA_IOTAM_DESCRIPTION;
+            delete process.env.IOTA_MONGO_HOST;
+            delete process.env.IOTA_MONGO_PORT;
+            delete process.env.IOTA_MONGO_DB;
+            delete process.env.IOTA_MONGO_REPLICASET;
+            delete process.env.IOTA_DEFAULT_RESOURCE;
+        });
+
+        afterEach(function(done) {
+            iotAgentLib.deactivate(done);
+        });
+
+        it('should not start and raise a MISSING_CONFIG_PARAMS error', function(done) {
+            iotAgentLib.activate(iotAgentConfig, function(error) {
+                config.getConfig().contextBroker.host.should.equal('cbhost');
+                config.getConfig().contextBroker.port.should.equal('1111');
+                config.getConfig().contextBroker.ngsiVersion.should.equal('v2');
```

Comment:
What happen if ngsiVersion was no defined in config.js or in env vars? Make sure ngsiVersion exists before check the value. I guess travis is failing due to that.

Reply:
ngsiVersion should be optional not a must in a config.js to ensure blackguard compatibility

Reply:
@AlvaroVega, as you can see in [the output of Travis](https://travis-ci.org/telefonicaid/iotagent-node-lib/builds/292080221?utm_source=github_status&utm_medium=notification), tests failing are not related to the new NGSIv2 features. It is a problem in the master branch.

I guess that you are having the same problems in PR #557

Reply:
I'm not an expet in Travis, but https://travis-ci.org/telefonicaid/iotagent-node-lib/branches is showing that master branch is ok. However, latest commit was 2 months ago. Maybe something has changed in that period and is causing the problem?

Maybe we shoul merge first PR #557 fixing travis stuff there before merging #556. Not sure the best way to go on...

Reply:
Yes, I think that before merging #556 the problem should be solved.

I have checked that there have been new releases of some of the dependencies. Maybe they are one of the causes?

Reply:
Could be. Dependencies in node are tricky and sometime in the past has been the cause of travis problems.